sources/sql/src/producer.rs

```diff
@@ -283,6 +285,37 @@ fn expr_to_sql(expr: &Expr, _schema: &DFSchemaRef, _col_ref_offset: usize) -> Re
     }
 }
 
+fn sort_to_sql(
+    sort_exprs: Vec<Expr>,
+    _schema: &DFSchemaRef,
+    _col_ref_offset: usize,
+) -> Result<Vec<OrderByExpr>, DataFusionError> {
+    let parsed_order_by_exprs = sort_exprs
+        .iter()
+        .map(|expr: &Expr| match expr {
+            Expr::Sort(sort_expr) => {
+                let col = expr_to_sql(&sort_expr.expr, _schema, _col_ref_offset)?;
+                Ok(OrderByExpr {
+                    asc: Some(sort_expr.asc),
+                    expr: col.clone(),
+                    nulls_first: Some(sort_expr.nulls_first),
+                })
+            }
+            _ => Err(DataFusionError::Plan("Expecting Sort expr".to_string())),
+        })
+        .collect::<Vec<Result<OrderByExpr, DataFusionError>>>();
```

Comment:
You can instead collect into: `Result<Vec<_>>` this allows you to handle any errors with a plain `?`. E..g.: `.collect::Result<Vec<_>>()?;`